.github/actions/generate-package-build-matrix/action.yml

```
@@ -0,0 +1,45 @@
+name: Composite action to generate the matrix of targets to build packages for.
```

Should this maybe have a simpler title? Like:
```suggestion
name: Generate target matrix
```